Update nuanced documentation tasks

dcalhoun opened the pull request:
This includes changes to documentation to ensure it aligns with our current release schedule, uses consistent names for repos throughout, and relies upon succinct, declarative statements in hopes of improving clarity overall. 

Each commit addresses a separate concern, with the `Refactor documentation sections` commit applying the largest and most aggressive refactor. It may be worthwhile reviewing them independently. We can revert the `Refactor documentation sections` commit if it is too aggressive.

dcalhoun committed:
Ensure placeholder values use same capitalization and format

The release process instructions state to find and replace all instances
of "X.XX.X", but inconsistency in capitalization can lead to missing a
placeholder.

dcalhoun committed:
Use consistent naming for repos

Alternating between proper nouns (e.g. "Gutenberg Mobile") and literal
repo names (e.g. "gutenberg-mobile") may cause confusion.

dcalhoun committed:
Simplify steps for syncing release to development branches

Rather than providing two avenues for syncing the release back (i.e.
`trunk` or `merge_release_X.XX.X_to_develop`), this provides a single
path in the hopes of improving clarity.

dcalhoun committed:
Refactor documentation sections

Update sections with consistent names in hopes of improving clarity
through declarative, succinct statements.

AmandaRiu commented on Releasing.md:
```diff
@@ -148,27 +159,27 @@ For the body of the post, just copy this checklist and again replace all occurre
 
 <!-- wp:paragraph -->
-<p>o Merge the gutenberg-mobile PR to <code>trunk</code>. WARNING: Don’t merge the gutenberg PR to <code>trunk</code> at this point.</p>
+<p>o Merge the gutenberg-mobile PR to <code>trunk</code>. WARNING: Do not merge the gutenberg PR into <code>trunk</code> at this point.</p>
```
I recommend bolding "gutenberg-mobile" and "gutenberg" in this paragraph just to make them stand out some more.

AmandaRiu reviewed (commented):
This is so much cleaner and easier to follow @dcalhoun! Thank you! I added a minor code comment and have the following suggestions as well:

### Create the Release

> Visit all opened PR's in gutenberg-mobile repo that are assigned to milestone X.XX.X and leave a message with options to (i) merge the PR as soon as possible, (ii) bump the PR to the next milestone, or (iii) remove the milestone from the PR.

It'd be nice to include a canned message for this section

> Message any related Slack channels to inform that the gutenberg-mobile release is now cut and any new WPiOS and WPAndroid changes having related gutenberg-mobile or gutenberg parts should now be merged to <code>gutenberg/after_X.XX.X</code> branches on WPiOS and WPAndroid until their own releases are cut next week.

It'd be nice to include a canned message along with the list of slack channels to post the message to for this line.


### Integrate the release section

> Make sure that the bundle files on the gutenberg-mobile release branch have been updated to include any changes to the release branch.

How would this be accomplished? Would it just be running `npm run bundle` in `gutenberg-mobile`? 

Regarding this section: 

![Screen Shot 2021-05-19 at 5 03 06 PM](https://user-images.githubusercontent.com/5810477/118885732-247ae000-b8c6-11eb-9b05-00ccc4dd35ed.png)


The above instructions say to create a tag in the **Gutenberg** repo, but then to create a new GitHub release in the **gutenberg-mobile** repo pointing to that tag. Am I reading that right? Can a release point to a tag not included in that repo? 

> In WPiOS, update the reference to point to the <em>tag</em>.

Which tag would WPiOS be pointing to? The **Gutenberg** tag mentioned a couple steps ago? It’d be helpful to add more specifics here. Also, is this done in a WPiOS PR? Which one? The one generated by the script? 

> In WPAndroid, update the <code>gutenbergMobileVersion</code> in <code>build.gradle</code> to point to the <em>tag</em>.

Same suggestion here as the one above regarding WPiOS.

> Once everything is merged, ping our friends in #platform9 and let them know we’ve merged our release so everything is right from our side to cut the main app releases. If the release is a beta/hot fix, be sure to directly mention the relevant Excellence Wranglers for the release.

It’d be nice to include a canned message here as well. 

That's all I have. I hope it helps. I tried to read it as someone who has never done a release before.

dcalhoun committed:
Clarify release checklist tasks

Address code review feedback by adding prepared messages and additional
details for nuanced tasks.

dcalhoun committed:
Treat GitHub's Release as a proper noun for clarity

Attempt to avoid confusion between the general "release" and the
creation of a "GitHub Release."

dcalhoun commented:
@AmandaRiu thank you for the thoughtful and helpful feedback! I have addressed each of your points. The rendered preview has been updated as well. 

> It'd be nice to include a canned message for this section

Absolutely agree. Great idea. I added prepared messages for each task requiring one. Let me know if you think any changes would be helpful. 

> > Make sure that the bundle files on the gutenberg-mobile release branch have been updated to include any changes to the release branch.
> 
> How would this be accomplished? Would it just be running `npm run bundle` in `gutenberg-mobile`?

Correct. I updated the task to clarify this. 

> ![Screen Shot 2021-05-19 at 5 03 06 PM](https://user-images.githubusercontent.com/5810477/118885732-247ae000-b8c6-11eb-9b05-00ccc4dd35ed.png)
> 
> The above instructions say to create a tag in the **Gutenberg** repo, but then to create a new GitHub release in the **gutenberg-mobile** repo pointing to that tag. Am I reading that right? Can a release point to a tag not included in that repo?

No, that reading is incorrect. The task intends to state that we should create a gutenberg-mobile Release based off the latest commit in gutenberg-mobile `trunk` (merge commit from previous task) which includes a `gutenberg` ref that points to the latest commit in the gutenberg release branch (which is also the `rnmobile/X.XX.X` tag created in a prior task). I have updated the tasks in the sections in an attempt to add more clarity. Let me know you what you think. 

> > In WPiOS, update the reference to point to the _tag_.
> 
> Which tag would WPiOS be pointing to? The **Gutenberg** tag mentioned a couple steps ago? It’d be helpful to add more specifics here. Also, is this done in a WPiOS PR? Which one? The one generated by the script?

I updated these tasks to be sequential and reference one another in hopes of clarifying _which_ tag is to be used. Let me if you think the changes are helpful. 

> That's all I have. I hope it helps. I tried to read it as someone who has never done a release before.

💯

dcalhoun committed:
Remove duplicative content

AmandaRiu commented:
I love these changes! I'm actually going to use this updated template for the 1.53.1 release to test drive it. 👍

AmandaRiu reviewed (commented):
Only one small change suggestion. What do you think about including a link to the [release instructions](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/develop/Releasing.md) to this task:

![Screen Shot 2021-05-20 at 9 50 49 PM](https://user-images.githubusercontent.com/5810477/119070322-c62d2a80-b9b5-11eb-8e58-de8f8c27dac0.png)

I know it seems a bit redundant since one would've had to have already visited this page to create the Release Scenario template, but if a different person is working the list it may be a nice nudge/reminder to prepare their branches before running the automation script. Maybe something similar to:

> In your clone of the release scripts, [review the instructions for running the script](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/develop/Releasing.md) via: `./release_automation.sh`. This creates the release branches and PRs for gutenberg-mobile, gutenberg, WPiOS, and WPAndroid. 

 WDYT?

dcalhoun committed:
Fix non-escaped HTMl disappearing

Copying and pasting the checklist into Gutenberg resulted in the
placeholder values disappearing, as they were considered HTML tags.
Using square brackets avoids this issue.

dcalhoun committed:
Add link to release script documentation

Provide a subtle bump to review the release script documentation prior
to running it.

dcalhoun committed:
Clarify differences between scheduled and beta/hot fix releases

Messaging differs depending on whether the current release is a
scheduled/normal release or a release to fix a regression in a frozen
branch.

dcalhoun committed:
Add task to perform a round of tests before cutting the release

A round of "smoke" tests are completed earlier in the week to identify
any breakages early.

dcalhoun committed:
Add general release timeline note

Provide loose schedule and links to current experiments.

mchowning commented on Releasing.md:
```diff
@@ -128,117 +164,130 @@ For the body of the post, just copy this checklist and again replace all occurre
 
 <!-- wp:paragraph -->
-<p>o Celebrate!</p>
+<p>o Pass the baton. Ping the dev who is responsible for the next release.</p>
```
I'm wondering if we should move this earlier (i.e. to before the next release wranglers should start smoke testing). Do you think it would be worth having the previous release wranglers "pass the baton" by doing something like posting a schedule for the upcoming release in Slack? I'm just trying to think what we can do to better make sure that the "new" release wranglers are aware of what they need to do, and I was thinking that the "previous" release wranglers probably know that better than anyone. Just a thought.

dcalhoun replied:
@mchowning absolutely. I believe we are actively running into that confusion now. The current rotation responsibilities actually last 3 weeks, with first week overlapping with the previous wrangler and the last week overlapping with the next wrangler.

AmandaRiu reviewed (approved):
Looks great @dcalhoun ! Ready to be merged

dcalhoun committed:
Add task to coordinate a schedule with the next release wrangler

The hope is that this will help avoid confusion regarding release
timelines.

dcalhoun committed:
Reframe provided messages as suggestions

A release wrangler should feel the freedom to craft their own message.
